src/AppBundle/Entity/Client.php

```diff
@@ -554,6 +554,19 @@ public function getUnsubmittedReports()
     }
 
     /**
+     * @JMS\VirtualProperty
+     * @JMS\Type("AppBundle\Entity\Report\Report")
+     * @JMS\SerializedName("current_report")
+     * @JMS\Groups({"current-report"})
+     *
+     * @return Report|null
+     */
+    public function getCurrentReport()
+    {
+        return $this->getUnsubmittedReports()->first() ?: null;
```

Comment:
Your solutions works for me. As a quick fix would something like this be relevant?
usort(
            $this->getUnsubmittedReports(),
            function ($a, $b) {
                return strtotime($b['submitted_date']) - strtotime($a['submitted_date']);
            }
        ); to reorder them.. then take the first record.